### PR TITLE
fix: [M-03] allow setting 21 bytes under LSP17Extension data key in LSP6KeyManager

### PR DESCRIPTION
**Pull request by YamenMerhi**
# What does this PR introduce?
## 🐛 Bug

- allow setting 21 bytes under LSP17Extension data key in LSP6KeyManager

### PR Checklist

- [ ] Wrote Tests
- [ ] Wrote & Generated Documentation (readme/natspec/dodoc)
- [ ] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [ ] Ran `npm run format` (prettier)
- [ ] Ran `npm run build`
- [ ] Ran `npm run test`

### CODE REVIEW

**Review by CJ42 — APPROVED** [review]
Should we test it with the other restricted controllers as well?

File: contracts/LSP6KeyManager/LSP6Modules/LSP6SetDataModule.sol [hunk 1]
```diff
@@ -322,7 +322,11 @@ abstract contract LSP6SetDataModule {
             // LSP17Extension:<bytes4>
         } else if (bytes12(inputDataKey) == _LSP17_EXTENSION_PREFIX) {
             // CHECK that `dataValue` contains exactly 20 bytes, which corresponds to an address for a LSP17 Extension
```
Comment (CJ42) [inline contracts/LSP6KeyManager/LSP6Modules/LSP6SetDataModule.sol:324]:
```suggestion
            // CHECK that `dataValue` contains exactly 20 or 21 bytes (if setting to forward value), which corresponds to an address for a LSP17 Extension
```

**Review by skimaharvey — APPROVED** [review]

**Review by b00ste — APPROVED** [review]

File: tests/LSP6KeyManager/Admin/PermissionChangeAddExtensions.test.ts [hunk 2]
```diff
@@ -1133,5 +1133,57 @@ export const shouldBehaveLikePermissionChangeOrAddExtensions = (
         });
       });
     });
+
+    describe('when setting random bytes under the LSP17Extension data key ', () => {
```
Comment (CJ42) [inline tests/LSP6KeyManager/Admin/PermissionChangeAddExtensions.test.ts:1137]: These tests are a copy paste of LSP20.
We are not in a LSP20 context here, we are calling via LSP6.

**Review by CJ42 — CHANGES_REQUESTED** [review]
We cannot write tests like that.
We need to check that validation happen also depending on the controller permission.

For instance, that regardless if controller has `ADDEXTENSION`, or `CHANGEEXTENSION`, or both, the error of invalid value being set occur.

**Review by CJ42 — APPROVED** [review]

**Pull request merged**